hooks/useOrders.ts

```diff
@@ -0,0 +1,72 @@
+import { useOrderHistory } from "@components/user/history/OrderHistoryProvider";
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react";
+import { useQuery, UseQueryResult } from "@tanstack/react-query";
+import { Order } from "models/Order";
+import { BoughtProduct, CartProduct } from "models/Product";
+import { Dispatch, SetStateAction } from "react";
+
+export const useOrders = (): UseQueryResult<Order[]> => {
+  const supabase = useSupabaseClient();
+  const user = useUser();
+  const { setOrders, setSelectedOrder } = useOrderHistory();
```

Comment:
I feel I am missing something. Why do `useOrderHistory` lives as state where `useOrders` consumes server data ?

Reply:
I don't know what you mean here. 
Should this be done another way?